core/src/main/java/hudson/model/Queue.java

```diff
@@ -675,6 +677,14 @@ public WaitingItem schedule(Task p, int quietPeriod, List<Action> actions) {
         }
     }
 
+    private void updateSnapshot() {
+        Snapshot revised = new Snapshot(waitingList, blockedProjects, buildables, pendings);
+        if (LOGGER.isLoggable(Level.FINEST)) {
```

Comment:
BTW you can also use the overload taking a `Supplier` lambda, which is more convenient.

Reply:
Ah but it looks like this method was just moved around? If so, better to return it to its original position, to minimize the diff.